persistent-template/Database/Persist/TH.hs

```diff
@@ -437,6 +438,8 @@ data MkPersistSettings = MkPersistSettings
     -- False.
     , mpsPrefixFields :: Bool
     -- ^ Prefix field names with the model name. Default: True.
+    , mpsFieldLabelModifier :: Text -> Text -> Text
+    -- ^ Customise the field accessors and lens names using the entity and field name. Default: appends both
```

Comment:
I'd prefer for things to just keep working as expected as much as possible. The corner case that you mention is that someone has `mpsPrefixFields = False`, *and* they go to use the `mpsFieldLabelModifier`. Obviously, they'd expect the entity name passed in with `mpsFieldLabelModifier` function, and not just empty string. 

I'm tempted to say that if they have specified `mpsPrefixFields = False`, then we just pass both things in to the function. The user is free to have a function like:

```haskell
mpsFieldLabelModifier = \entity field -> "_schema_x_" <> field
```

With this design, we don't know if the user has *provided* an implementation or is using the default. So perhaps we should set the type here to `mpsFieldLabelModifier ::Maybe (Text -> Text -> Text)` and a default of `Nothing`.

Then the logic looks like:

```haskell
case mpsFieldLabelModifier mps of
  Nothing -> do the logic as currently written
  Just fn -> provide both entity name and field to the function
```

This basically makes `mpsPrefixFields` redundant, in the case that the user specifies `mpsPrefixFields = False, mpsFieldLabelModifier = Just ...`. 

Another thing we'll want to consider: if they have `mpsPrefixFields = False`, do we want to uppercase the first character of the field name for the user? Whatever choice we make should be documented.

Reply:
I've thought about this some more. I make two observations:

1. We want the change to be backwards compatible for users using `mpsPrefixFields = False` who don't want to use the `mpsFieldLabelModifier`
2. Other than for this backward compatibility case, `mpsPrefixFields` is redundant since you can discard `entity` when specifying the `mpsFieldLabelModifier`.

I propose that we deprecate `mpsPrefixFields` and document that `mpsFieldLabelModifier` will be ignored if `mpsPrefixFields` is set to False. 
This way, we avoid complicating the type of the fieldLabelModifier.

Reply:
That sounds good to me!

Reply:
Observation 2 is not completely correct. mpsPrefixFields is also used for disabling the entity prefix in the constraint name, which should be uppercase and not start with an underscore.
One solution is to reuse the modifier function, drop any leading underscore and uppercase the result, but maybe an additional modifier function for the constraints would make more sense?

Reply:
I went ahead and implemented a second modifier function (mpsConstraintLabelModifier) for the constraints which need to be upper case. Let me know what you think